conans/model/conan_file.py

```diff
@@ -50,13 +50,13 @@ def create_requirements(conanfile):
         raise ConanException("Error while initializing requirements. %s" % str(e))
 
 
-def create_settings(conanfile, settings, local):
+def create_settings(conanfile, settings):
     try:
         defined_settings = getattr(conanfile, "settings", None)
         if isinstance(defined_settings, str):
             defined_settings = [defined_settings]
         current = defined_settings or {}
-        settings.constraint(current, raise_undefined_field=not local)
+        settings.constraint(current)
```

Comment:
That means the settings will be always complete now?

Reply:
Yes (or it should fail otherwise), because ``configure()`` will be called.